app/controllers/result_filters/location_controller.rb

```diff
@@ -19,20 +19,22 @@ def create
 
       form_params = strip(filter_params.clone).merge(sortby: ResultsView::DISTANCE)
       form_object = LocationFilterForm.new(form_params)
+
       if form_object.valid?
-        all_params = form_params.merge!(form_object.params)
-        redirect_to(next_step(all_params))
+        parameters_with_geocode_added_and_previous_removed = remove_previous_parameters(form_params.merge!(form_object.params))
```

Comment:
Is the `merge!` here intentional? Does `form_params` get used in the view or something?

Reply:
I don't think it's needed. I've removed it.